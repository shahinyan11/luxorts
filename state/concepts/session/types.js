const namespace = 'session';

export const USER_SIGN_UP = `${namespace}/USER_SIGN_UP`;
export const USER_SIGN_IN_SUCCESS = `${namespace}/USER_SIGN_IN_SUCCESS`;
export const USER_SIGN_OUT = `${namespace}/USER_SIGN_OUT`;
export const USER_SIGN_OUT_SUCCESS = `${namespace}/USER_SIGN_OUT_SUCCESS`;
export const VERIFY_USER_EMAIL = `${namespace}/VERIFY_USER_EMAIL`;
export const VERIFY_USER_PHONE = `${namespace}/VERIFY_USER_PHONE`;
export const RESEND_USER_EMAIL_VERIFICATION = `${namespace}/RESEND_USER_EMAIL_VERIFICATION`;
export const USER_SIGN_IN = `${namespace}/USER_SIGN_IN`;
export const USER_ACTIVATE_ACCOUNT = `${namespace}/USER_ACTIVATE_ACCOUNT`;
export const UPDATE_USER = `${namespace}/UPDATE_USER`;
export const USER_SEND_FORGOT_PASSWORD_EMAIL = `${namespace}/USER_SEND_FORGOT_PASSWORD_EMAIL`;
export const USER_RESEND_FORGOT_PASSWORD_EMAIL = `${namespace}/USER_RESEND_FORGOT_PASSWORD_EMAIL`;
export const USER_VALIDATE_FORGOT_PASSWORD_CODE = `${namespace}/USER_VALIDATE_FORGOT_PASSWORD_CODE`;
export const USER_UPDATE_PASSWORD = `${namespace}/USER_UPDATE_PASSWORD`;
export const USER_UPDATE_AVATAR = `${namespace}/USER_UPDATE_AVATAR`;
export const USER_REMOVE_AVATAR = `${namespace}/USER_REMOVE_AVATAR`;
export const USER_UPDATE_PROFILE = `${namespace}/USER_UPDATE_PROFILE`;
export const USER_DEACTIVATE_ACCOUNT = `${namespace}/USER_DEACTIVATE_ACCOUNT`;
export const USER_CHANGE_PASSWORD = `${namespace}/USER_CHANGE_PASSWORD`;
export const FETCH_USER_INVITATION = `${namespace}/FETCH_USER_INVITATION`;
export const SET_INVITATIONS_PAGE = `${namespace}/SET_INVITATIONS_PAGE`;
export const USER_INVITE_FRIENDS = `${namespace}/USER_INVITE_FRIENDS`;
export const USER_RESEND_INVITE = `${namespace}/USER_RESEND_INVITE`;
export const USER_SEND_UPDATE_PHONE = `${namespace}/USER_SEND_UPDATE_PHONE`;
export const USER_RESEND_UPDATE_PHONE = `${namespace}/USER_RESEND_UPDATE_PHONE`;
export const USER_UPDATE_PHONE = `${namespace}/USER_UPDATE_PHONE`;
export const USER_SEND_UPDATE_EMAIL = `${namespace}/USER_SEND_UPDATE_EMAIL`;
export const USER_RESEND_UPDATE_EMAIL = `${namespace}/USER_RESEND_UPDATE_EMAIL`;
export const USER_UPDATE_EMAIL = `${namespace}/USER_UPDATE_EMAIL`;
