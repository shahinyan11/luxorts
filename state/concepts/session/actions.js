import makeFormSubmitAction from 'utils/form/makeFormSubmitAction';
import * as types from 'state/concepts/session/types';

export const userSignUp = makeFormSubmitAction(types.USER_SIGN_UP);

export const userSignInSuccess = (currentUser) => ({
  type: types.USER_SIGN_IN_SUCCESS,
  currentUser,
});

export const userSignOut = () => ({
  type: types.USER_SIGN_OUT,
});

export const userSignOutSuccess = () => ({
  type: types.USER_SIGN_OUT_SUCCESS,
});

export const verifyUserEmail = makeFormSubmitAction(types.VERIFY_USER_EMAIL);

export const verifyUserPhone = makeFormSubmitAction(types.VERIFY_USER_PHONE);

export const resendUserEmailVerification = () => ({
  type: types.RESEND_USER_EMAIL_VERIFICATION,
});

export const userSignIn = makeFormSubmitAction(types.USER_SIGN_IN);

export const userActivateAccount = ({ redirectRoute = null, isModal = false }) => ({
  type: types.USER_ACTIVATE_ACCOUNT,
  redirectRoute,
  isModal,
});

export const updateUser = (userData) => ({
  type: types.UPDATE_USER,
  userData,
});

export const userSendForgotPasswordEmail = makeFormSubmitAction(
  types.USER_SEND_FORGOT_PASSWORD_EMAIL,
);

export const userValidateForgotPasswordCode = makeFormSubmitAction(
  types.USER_VALIDATE_FORGOT_PASSWORD_CODE,
);

export const userResendForgotPasswordEmail = () => ({
  type: types.USER_RESEND_FORGOT_PASSWORD_EMAIL,
});

export const userUpdatePassword = makeFormSubmitAction(types.USER_UPDATE_PASSWORD);

export const userUpdateAvatar = (file) => ({
  type: types.USER_UPDATE_AVATAR,
  file,
});

export const userRemoveAvatar = () => ({
  type: types.USER_REMOVE_AVATAR,
});

export const userUpdateProfile = makeFormSubmitAction(types.USER_UPDATE_PROFILE);

export const userDeactivateAccount = makeFormSubmitAction(types.USER_DEACTIVATE_ACCOUNT);

export const userChangePassword = makeFormSubmitAction(types.USER_CHANGE_PASSWORD);

export const userInviteFriends = makeFormSubmitAction(types.USER_INVITE_FRIENDS);

export const fetchUserInvitation = () => ({
  type: types.FETCH_USER_INVITATION,
});

export const setInvitationsPage = (pageNumber) => ({
  type: types.SET_INVITATIONS_PAGE,
  pageNumber,
});

export const userResendInvite = (invitationId) => ({
  type: types.USER_RESEND_INVITE,
  invitationId,
});

export const userSendUpdatePhone = makeFormSubmitAction(types.USER_SEND_UPDATE_PHONE);

export const userResendUpdatePhone = () => ({
  type: types.USER_RESEND_UPDATE_PHONE,
});

export const userUpdatePhone = makeFormSubmitAction(types.USER_UPDATE_PHONE);

export const userResendUpdateEmail = () => ({
  type: types.USER_RESEND_UPDATE_EMAIL,
});

export const userSendUpdateEmail = makeFormSubmitAction(types.USER_SEND_UPDATE_EMAIL);

export const userUpdateEmail = makeFormSubmitAction(types.USER_UPDATE_EMAIL);
