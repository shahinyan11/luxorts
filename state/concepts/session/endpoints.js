import endpoint from 'utils/endpoint';
import {
  userAccountActivationRoute,
  userAccountDeactivationRoute,
  userAccountForgotPasswordCodeRoute,
  userAccountForgotPasswordResendRoute,
  userAccountForgotPasswordRoute,
  userChangePasswordRoute,
  userInvitationsRoute,
  userRegistrationRoute,
  userResendInviteRoute,
  userResendUpdateEmailRoute,
  userResendUpdatePhoneRoute,
  userSessionRoute,
  userUpdateAvatarRoute,
  userUpdateEmailRoute,
  userUpdatePhoneRoute,
  userUpdateProfileRoute,
  userVerificationEmailRoute,
  userVerificationPhoneRoute,
} from 'lib/apiRoutes';

import * as types from './types';

export const createUserEndpoint = endpoint(types.USER_SIGN_UP, 'POST', userRegistrationRoute);

export const verifyUserEmailEndpoint = endpoint(
  types.VERIFY_USER_EMAIL,
  'POST',
  userVerificationEmailRoute,
);

export const verifyUserPhoneEndpoint = endpoint(
  types.VERIFY_USER_PHONE,
  'POST',
  userVerificationPhoneRoute,
);

export const resendUserEmailVerificationEndpoint = endpoint(
  types.RESEND_USER_EMAIL_VERIFICATION,
  'PUT',
  userVerificationEmailRoute,
);

export const signInUserEndpoint = endpoint(types.USER_SIGN_IN, 'POST', userSessionRoute);

export const userAccountActivationEndpoint = endpoint(
  types.USER_SIGN_IN,
  'POST',
  userAccountActivationRoute,
);

export const signOutUserEndpoint = endpoint(types.USER_SIGN_OUT, 'DELETE', userSessionRoute);

export const userSendForgotPasswordEmailEndpoint = endpoint(
  types.USER_SEND_FORGOT_PASSWORD_EMAIL,
  'POST',
  userAccountForgotPasswordRoute,
);

export const userValidateForgotPasswordCodeEndpoint = endpoint(
  types.USER_VALIDATE_FORGOT_PASSWORD_CODE,
  'POST',
  userAccountForgotPasswordCodeRoute,
);

export const userResendForgotPasswordEmailEndpoint = endpoint(
  types.USER_RESEND_FORGOT_PASSWORD_EMAIL,
  'POST',
  userAccountForgotPasswordResendRoute,
);

export const userUpdatePasswordEndpoint = endpoint(
  types.USER_UPDATE_PASSWORD,
  'PUT',
  userAccountForgotPasswordRoute,
);

export const userUpdateAvatarEndpoint = endpoint(
  types.USER_UPDATE_AVATAR,
  'PUT',
  userUpdateAvatarRoute,
);

export const userRemoveAvatarEndpoint = endpoint(
  types.USER_REMOVE_AVATAR,
  'DELETE',
  userUpdateAvatarRoute,
);

export const userUpdateProfileEndpoint = endpoint(
  types.USER_UPDATE_PROFILE,
  'PUT',
  userUpdateProfileRoute,
);

export const userAccountDeactivationEndpoint = endpoint(
  types.USER_DEACTIVATE_ACCOUNT,
  'POST',
  userAccountDeactivationRoute,
);

export const userChangePasswordEndpoint = endpoint(
  types.USER_CHANGE_PASSWORD,
  'PUT',
  userChangePasswordRoute,
);

export const userInviteFriendsEndpoint = endpoint(
  types.USER_INVITE_FRIENDS,
  'POST',
  userInvitationsRoute,
);

export const fetchUserInvitationEndpoint = endpoint(
  types.FETCH_USER_INVITATION,
  'GET',
  userInvitationsRoute,
);

export const userResendInviteEndpoint = (invitationId) =>
  endpoint(types.USER_RESEND_INVITE, 'PUT', userResendInviteRoute(invitationId));

export const userSendUpdatePhoneEndpoint = endpoint(
  types.USER_SEND_UPDATE_PHONE,
  'POST',
  userUpdatePhoneRoute,
);

export const userResendUpdatePhoneEndpoint = endpoint(
  types.USER_RESEND_UPDATE_PHONE,
  'POST',
  userResendUpdatePhoneRoute,
);

export const userUpdatePhoneEndpoint = endpoint(
  types.USER_UPDATE_PHONE,
  'PUT',
  userUpdatePhoneRoute,
);

export const userSendUpdateEmailEndpoint = endpoint(
  types.USER_SEND_UPDATE_EMAIL,
  'POST',
  userUpdateEmailRoute,
);

export const userResendUpdateEmailEndpoint = endpoint(
  types.USER_RESEND_UPDATE_EMAIL,
  'POST',
  userResendUpdateEmailRoute,
);

export const userUpdateEmailEndpoint = endpoint(
  types.USER_UPDATE_EMAIL,
  'PUT',
  userUpdateEmailRoute,
);
