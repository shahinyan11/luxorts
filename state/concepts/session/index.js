import sessionReducer from 'state/concepts/session/reducer';

export { default as sessionOperations } from 'state/concepts/session/operations';
export * as sessionTypes from 'state/concepts/session/types';
export default sessionReducer;
