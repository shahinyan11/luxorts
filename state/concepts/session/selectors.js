import { createSelector } from 'reselect';
import { path, prop } from 'ramda';
import build from 'redux-object';

import isPresent from 'utils/isPresent';

import {
  fetchUserInvitationEndpoint,
  userSendUpdatePhoneEndpoint,
} from 'state/concepts/session/endpoints';
import { newPhoneSelector, totalCountSelector } from 'state/data/selectors';

const dataSelector = prop('data');

export const currentUserIdSelector = path(['session', 'currentUser', 'id']);

export const currentUserEmailSelector = path(['session', 'currentUser', 'email']);

export const currentUserSelector = path(['session', 'currentUser']);

export const isUserLoggedInSelector = createSelector(currentUserIdSelector, isPresent);

export const userInvitationSelector = createSelector(
  dataSelector,
  (data) => build(data, 'userInvitation') || [],
);

export const userNewPhoneSelector = (state) =>
  newPhoneSelector(state, userSendUpdatePhoneEndpoint.endpoint);

export const paginationSelector = path(['session', 'pagination']);

export const userInvitationCountSelector = (state) =>
  totalCountSelector(state, fetchUserInvitationEndpoint.endpoint);
