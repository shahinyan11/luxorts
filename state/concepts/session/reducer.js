import { combineReducers } from 'redux';
import { mergeDeepRight } from 'ramda';

import { PAGE } from 'constants';

import * as types from 'state/concepts/session/types';

const currentUser = (state = null, action) => {
  switch (action.type) {
    case types.USER_SIGN_IN_SUCCESS:
      return action.currentUser;
    case types.UPDATE_USER:
      return mergeDeepRight(state, action.userData);
    case types.USER_SIGN_OUT_SUCCESS:
      return null;
    default:
      return state;
  }
};

const pagination = (state = PAGE, action) => {
  switch (action.type) {
    case types.SET_INVITATIONS_PAGE:
      return { ...state, number: action.pageNumber };
    default:
      return state;
  }
};

const sessionReducer = combineReducers({
  currentUser,
  pagination,
});

export default sessionReducer;
