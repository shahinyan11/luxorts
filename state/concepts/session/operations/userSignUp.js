import { createLogic } from 'redux-logic';
import normalize from 'json-api-normalizer';
import build from 'redux-object';
import { camelCase } from 'lodash';

import redirect from 'utils/redirect';
import assignFormErrors from 'utils/form/assignFormErrors';
import setCookieIsomorphic from 'utils/setCookieIsomorphic';
import setAuthorizationHeader from 'utils/setAuthorizationHeader';

import { USER_SIGN_UP } from 'state/concepts/session/types';
import { dataApiFailure, dataApiRequest, dataApiSuccess } from 'state/data/actions';
import { userSignInSuccess } from 'state/concepts/session/actions';
import { createUserEndpoint } from 'state/concepts/session/endpoints';

const userSignUp = createLogic({
  type: USER_SIGN_UP,
  latest: true,

  async process({ action: { values, form }, httpClient }, dispatch, done) {
    const { endpoint, url } = createUserEndpoint;

    dispatch(dataApiRequest({ endpoint }));
    try {
      const body = {
        email: values.email,
        password: values.password,
        invitation_id: values.invitationID,
        user_profile: {
          first_name: values.userProfile.firstName,
          last_name: values.userProfile.lastName,
          phone_number: values.userProfile.phoneNumber,
          date_of_birth: values.userProfile.dateOfBirth,
        },
      };

      const { data } = await httpClient.post(url, body);
      const response = normalize(data);
      const currentUser = build(response, camelCase(data.data.type), data.data.id);

      setCookieIsomorphic(null, 'currentUser', JSON.stringify(currentUser));
      setCookieIsomorphic(null, 'tokens', JSON.stringify(data.meta));
      setAuthorizationHeader(httpClient, data.meta.access);

      dispatch(userSignInSuccess(currentUser));
      dispatch(dataApiSuccess({ response, endpoint }));

      if (values.redirectRoute) {
        redirect(values.redirectRoute);
      }
    } catch (error) {
      assignFormErrors(form, error);
      dispatch(dataApiFailure({ endpoint }));
    }

    done();
  },
});

export default userSignUp;
