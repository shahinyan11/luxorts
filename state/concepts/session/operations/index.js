import userSignUp from './userSignUp';
import verifyUserEmail from './verifyUserEmail';
import resendUserEmailVerification from './resendUserEmailVerification';
import userSignIn from './userSignIn';
import userActivateAccount from './userActivateAccount';
import userSignOut from './userSignOut';
import userSendForgotPasswordEmail from './userSendForgotPasswordEmail';
import userValidateForgotPasswordCode from './userValidateForgotPasswordCode';
import userResendForgotPasswordEmail from './userResendForgotPasswordEmail';
import userUpdatePassword from './userUpdatePassword';
import verifyUserPhone from './verifyUserPhone';
import updateUser from './updateUser';
import userUpdateAvatar from './userUpdateAvatar';
import userRemoveAvatar from './userRemoveAvatar';
import userUpdateProfile from './userUpdateProfile';
import userDeactivateAccount from './userDeactivateAccount';
import userChangePassword from './userChangePassword';
import userInviteFriends from './userInviteFriends';
import fetchUserInvitation from './fetchUserInvitation';
import userResendInvite from './userResendInvite';
import userSendUpdatePhone from './userSendUpdatePhone';
import userResendUpdatePhone from './userResendUpdatePhone';
import userUpdatePhone from './userUpdatePhone';
import userSendUpdateEmail from './userSendUpdateEmail';
import userResendUpdateEmail from './userResendUpdateEmail';
import userUpdateEmail from './userUpdateEmail';

export default [
  userSignUp,
  verifyUserEmail,
  resendUserEmailVerification,
  userSignIn,
  userActivateAccount,
  userSignOut,
  userSendForgotPasswordEmail,
  userValidateForgotPasswordCode,
  userResendForgotPasswordEmail,
  userUpdatePassword,
  verifyUserPhone,
  updateUser,
  userUpdateAvatar,
  userRemoveAvatar,
  userUpdateProfile,
  userDeactivateAccount,
  userChangePassword,
  userInviteFriends,
  fetchUserInvitation,
  userResendInvite,
  userSendUpdatePhone,
  userResendUpdatePhone,
  userUpdatePhone,
  userSendUpdateEmail,
  userResendUpdateEmail,
  userUpdateEmail,
];
