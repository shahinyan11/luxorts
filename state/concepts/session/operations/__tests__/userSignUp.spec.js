import normalize from 'json-api-normalizer';
import build from 'redux-object';
import { omit, pick } from 'ramda';

import setCookieIsomorphic from 'utils/setCookieIsomorphic';
import mockHttpClient from 'utils/testHelpers/mockHttpClient';
import formatJsonApiErrors from 'utils/formatJsonApiErrors';
import redirect from 'utils/redirect';

import { userSignInSuccess } from 'state/concepts/session/actions';
import { dataApiRequest, dataApiSuccess, dataApiFailure } from 'state/data/actions';
import { createUserEndpoint } from 'state/concepts/session/endpoints';

import {
  loginSuccessResponse,
  loginErrorResponse,
} from 'state/concepts/session/__mocks__/loginResponse';
import userSignUp from '../userSignUp';

jest.mock('utils/setCookieIsomorphic');
jest.mock('utils/redirect');

describe('userSignUp', () => {
  let dispatch;

  const action = {
    values: {
      email: 'email',
      password: 'password',
      invitationID: 'invitationID',
      redirectRoute: 'redirectRoute',
      userProfile: {
        firstName: 'firstName',
        lastName: 'lastName',
        phoneNumber: 'phoneNumber',
        dateOfBirth: 'dateOfBirth',
      },
    },
    form: {
      setErrors: jest.fn(),
      setStatus: jest.fn(),
    },
  };

  const body = {
    email: action.values.email,
    password: action.values.password,
    invitation_id: action.values.invitationID,
    user_profile: {
      first_name: action.values.userProfile.firstName,
      last_name: action.values.userProfile.lastName,
      phone_number: action.values.userProfile.phoneNumber,
      date_of_birth: action.values.userProfile.dateOfBirth,
    },
  };

  const { endpoint, url } = createUserEndpoint;

  const beforeFunction = (httpClient) => () => {
    jest.clearAllMocks();
    dispatch = jest.fn();
    userSignUp.process({ httpClient, action }, dispatch, jest.fn());
  };

  it('matches snapshot', () => {
    expect(userSignUp).toMatchSnapshot();
  });

  describe('success', () => {
    const httpClient = mockHttpClient({ method: 'post', response: loginSuccessResponse });

    httpClient.defaults = { headers: { common: {} } };

    const response = normalize(loginSuccessResponse.data);

    const currentUser = build(response, 'travellerAccount', loginSuccessResponse.data.data.id);

    beforeEach(beforeFunction(httpClient));

    it('calls right endpoint with right params', () => {
      expect(httpClient.post).toHaveBeenCalledWith(url, body);
    });

    it('dispatches actions', () => {
      expect(dispatch).toHaveBeenCalledTimes(3);

      expect(dispatch).toHaveBeenNthCalledWith(1, dataApiRequest({ endpoint }));
      expect(dispatch).toHaveBeenNthCalledWith(2, userSignInSuccess(currentUser));
      expect(dispatch).toHaveBeenNthCalledWith(
        3,
        dataApiSuccess({
          endpoint,
          response,
        }),
      );
    });

    it('sets tokens', () => {
      const tokens = loginSuccessResponse.data.meta;

      expect(setCookieIsomorphic).toHaveBeenCalledTimes(2);

      expect(setCookieIsomorphic).toHaveBeenNthCalledWith(
        1,
        null,
        'currentUser',
        JSON.stringify(currentUser),
      );
      expect(setCookieIsomorphic).toHaveBeenNthCalledWith(
        2,
        null,
        'tokens',
        JSON.stringify(tokens),
      );
      expect(httpClient.defaults.headers.common.Authorization).toBe(`Bearer ${tokens.access}`);
    });

    it('redirects', () => {
      expect(redirect).toHaveBeenCalledWith(action.values.redirectRoute);
    });
  });

  describe('failure', () => {
    const httpClient = mockHttpClient({
      method: 'post',
      response: loginErrorResponse,
      reject: true,
    });

    beforeEach(beforeFunction(httpClient));

    it('dispatches actions', () => {
      expect(dispatch).toHaveBeenCalledTimes(2);

      expect(dispatch).toHaveBeenNthCalledWith(1, dataApiRequest({ endpoint }));
      expect(dispatch).toHaveBeenNthCalledWith(2, dataApiFailure({ endpoint }));
    });

    it('sets errors', () => {
      const formattedErrors = formatJsonApiErrors(loginErrorResponse.response.data.errors);

      expect(action.form.setStatus).toHaveBeenCalledWith(pick(['base'], formattedErrors));
      expect(action.form.setErrors).toHaveBeenCalledWith(omit(['base'], formattedErrors));
    });
  });
});
