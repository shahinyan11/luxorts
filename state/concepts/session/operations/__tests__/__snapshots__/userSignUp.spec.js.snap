// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userSignUp matches snapshot 1`] = `
Object {
  "cancelType": undefined,
  "debounce": 0,
  "latest": true,
  "name": undefined,
  "process": [Function],
  "processOptions": Object {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "session/USER_SIGN_UP",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
